#! /usr/bin/env python

PACKAGE='tuw_self_localization'
from dynamic_reconfigure.parameter_generator_catkin import *
gen = ParameterGenerator()

gen.add("reinitialize", bool_t,   0, "",  False)
gen.add("initial_with_ground_truth",    bool_t,   0, "",  False)
gen.add("plot_data",    bool_t,   0, "",  False)
gen.add("map_pix_x", int_t, 0, "map size in pixel", 558, 10, 2000)
gen.add("map_pix_y", int_t, 0, "map size in pixel", 558, 10, 2000)
gen.add("map_min_x", double_t, 0, "map size in meter",-9, -20, 20)
gen.add("map_max_x", double_t, 0, "map size in meter", 9, -20, 20)
gen.add("map_min_y", double_t, 0, "map size in meter",-9, -20, 20)
gen.add("map_max_y", double_t, 0, "map size in meter", 9, -20, 20)
gen.add("map_grid_x", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_grid_y", double_t, 0, "grid size in meter", 1, 0, 10)
gen.add("map_rotation", double_t, 0, "map size in meter", 0, -3.14, 3.14)



exit(gen.generate(PACKAGE, "tuw_self_localization", "SelfLocalization"))
